Modules/ITS/src/ITSDecodingErrorTask.cxx

```diff
@@ -0,0 +1,210 @@
+      mChipErrorCount[ifee][ierror] = 0;
+    }
+  }
+}
+
+void ITSDecodingErrorTask::createDecodingPlots()
+{
+  mLinkErrorVsFeeid = new TH2I("General/LinkErrorVsFeeid", "GBTLink errors per FeeId", NFees, 0, NFees, o2::itsmft::GBTLinkDecodingStat::NErrorsDefined, 0.5, o2::itsmft::GBTLinkDecodingStat::NErrorsDefined + 0.5);
+  mLinkErrorVsFeeid->SetMinimum(0);
+  mLinkErrorVsFeeid->SetStats(0);
+  getObjectsManager()->startPublishing(mLinkErrorVsFeeid);
+  mChipErrorVsFeeid = new TH2I("General/ChipErrorVsFeeid", "Chip decoding errors per FeeId", NFees, 0, NFees, o2::itsmft::ChipStat::NErrorsDefined, 0.5, o2::itsmft::ChipStat::NErrorsDefined + 0.5);
+  mChipErrorVsFeeid->SetMinimum(0);
+  mChipErrorVsFeeid->SetStats(0);
+  getObjectsManager()->startPublishing(mChipErrorVsFeeid);
+  mLinkErrorPlots = new TH1D("General/LinkErrorPlots", "GBTLink decoding Errors", o2::itsmft::GBTLinkDecodingStat::NErrorsDefined, 0.5, o2::itsmft::GBTLinkDecodingStat::NErrorsDefined + 0.5);
+  mLinkErrorPlots->SetMinimum(0);
+  mLinkErrorPlots->SetFillColor(kRed);
+  getObjectsManager()->startPublishing(mLinkErrorPlots); // mLinkErrorPlots
+  mChipErrorPlots = new TH1D("General/ChipErrorPlots", "Chip Decoding Errors", o2::itsmft::ChipStat::NErrorsDefined, 0.5, o2::itsmft::ChipStat::NErrorsDefined + 0.5);
+  mChipErrorPlots->SetMinimum(0);
+  mChipErrorPlots->SetFillColor(kRed);
+  getObjectsManager()->startPublishing(mChipErrorPlots); // mChipErrorPlots
+}
+
+void ITSDecodingErrorTask::setAxisTitle(TH1* object, const char* xTitle, const char* yTitle)
+{
+  object->GetXaxis()->SetTitle(xTitle);
+  object->GetYaxis()->SetTitle(yTitle);
+}
+
+void ITSDecodingErrorTask::setPlotsFormat()
+{
+  if (mLinkErrorVsFeeid) {
+    setAxisTitle(mLinkErrorVsFeeid, "FeeID", "Error ID");
+  }
+  if (mChipErrorVsFeeid) {
+    setAxisTitle(mChipErrorVsFeeid, "FeeID", "Error ID");
+  }
+  if (mLinkErrorPlots) {
+    setAxisTitle(mLinkErrorPlots, "LinkError ID", "Counts");
+  }
+  if (mChipErrorPlots) {
+    setAxisTitle(mChipErrorPlots, "ChipError ID", "Counts");
+  }
+}
+
+void ITSDecodingErrorTask::startOfActivity(Activity& activity)
+{
+  ILOG(Info, Support) << "startOfActivity : " << activity.mId << ENDM;
+  mRunNumber = activity.mId;
+}
+
+void ITSDecodingErrorTask::startOfCycle() { ILOG(Info, Support) << "startOfCycle" << ENDM; }
+
+void ITSDecodingErrorTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  // set timer
+  //
+  std::chrono::time_point<std::chrono::high_resolution_clock> start;
+  std::chrono::time_point<std::chrono::high_resolution_clock> end;
+  // int difference;
+  start = std::chrono::high_resolution_clock::now();
+
+  std::vector<InputSpec> rawDataFilter{ InputSpec{ "", ConcreteDataTypeMatcher{ "DS", "RAWDATA0" }, Lifetime::Timeframe } };
+
+  rawDataFilter.push_back(InputSpec{ "", ConcreteDataTypeMatcher{ "ITS", "RAWDATA" }, Lifetime::Timeframe });
+  DPLRawParser parser(ctx.inputs(), rawDataFilter);
```

Comment:
You don't need rawDataFilter and parser since you receive all information from "linkerrors" and "decerrors"

Reply:
ok, I get it